src/bench.c

```diff
@@ -799,9 +799,21 @@ int benchmark_all(void)
 		if ((format->params.flags & FMT_DYNAMIC) ||
 		    strstr(format->params.label, "-opencl") ||
 		    strstr(format->params.label, "-ztex") ||
-		    !strcmp(format->params.label, "crypt"))
+		    !strcmp(format->params.label, "crypt")) {
+#if defined(HAVE_OPENCL)
+			if (strstr(format->params.label, "-opencl")) {
+				/* Allow OpenCL build's "--test" to run on no-OpenCL systems */
+				/* It is a like a hack, but it is necessary since OpenCL must be started */
+				/* to get the number of devices. The init[1] itself is performed only once. */
+				/* [1] Enumeration of platforms and devices; and parse options. */
+				opencl_load_environment();
+
+				if (get_number_of_available_devices() == 0)
+					continue;
+			}
+#endif
 			fmt_init(format);
-
+		}
```

Comment:
I think we need an empty line after the closing curly brace, to keep separation of this logical block from the next (OpenCL vs. mask mode).